packages/edit-site/src/components/header/index.js

```diff
@@ -56,6 +67,33 @@ export default function Header( { openEntitiesSavedStates } ) {
 			} ) ),
 		[]
 	);
+	const setActivePage = useCallback( async ( newPage ) => {
+		try {
+			const { success, data } = await fetch(
```

Comment:
Just stumbled upon this and it seems a bit hacky. What does this fetch do? Why not core-data? Why not apiFetch?

Reply:
Also, it seems to me that all the logic here is better on an async store action. (generator with controls) as otherwise there's a lot of potential for breakage as written (set state on unmounted components).

Reply:
I guess  the issue is that "settings" is not yet on a store, for me, this is becoming more and more a priority as we add components and share state.

Reply:
It's not precisely an API endpoint. It's the only straightforward way of triggering Core's template resolution for a URL because calling `parse_request` in WP twice is not trivial.

Yes, I agree that at this point, it makes sense to move settings into a store and have this be an action.

Reply:
I'm also wondering if we're making things hard for us by trying to act like a full  SPA here where we could instead trigger a reload and append something to the URL? (The dirty detection check could help us to avoid leaving the page with unsaved changes).

Reply:
We could add deep linking without refreshes too.

I do think that linking to a specific page in the site editor from different places would be nice.

Reply:
> I'm also wondering if we're making things hard for us by trying to act like a full SPA here where we could instead trigger a reload and append something to the URL?

IMO, it would be nice to avoid hard refreshes. One of the biggest user complaints on WordPress.com is how long the block editor takes to load, which is a combination of loading an iframe and loading the block editor. Destroying the existing JS context and rebuilding it seems to be really inefficient (IMO) when the code can already handle switching pages client side! Since switching pages and templates seems to be a common task, it’d be nice to keep that interaction quick :)